src/app/shared/deviceSniffingHelper.js

```diff
@@ -0,0 +1,37 @@
+const UAParser = require("ua-parser-js");
+const PHONE_TYPES = require("../../constants/phone-types");
+const OS_TYPES = require("../../constants/os-types");
+
+function modifyJourneyOnSniffing(req) {
+  const parser = new UAParser(req.headers["user-agent"]);
+
+  if (parser.getDevice()["type"] === "mobile") {
+    req.body.journey += "Smartphone";
+    switch (parser.getOS()["name"]) {
+      case OS_TYPES.IOS:
+        req.body.journey += "Iphone";
+        break;
+      case OS_TYPES.ANDROID:
+        req.body.journey += "Android";
+        break;
+    }
+  }
+}
+
+function preferenceSniffedPhoneType(req) {
```

Comment:
Happy to discuss this, but I think this method would be easier to understand if it was called something like `sniffPhoneType` and took `req.params.specifiedPhoneType` as a parameter called `fallback` or similar.

Reply:
I like your suggestion! Will go with it